manager/logbroker/broker.go

```diff
@@ -383,6 +383,15 @@ func (lb *LogBroker) PublishLogs(stream api.LogBroker_PublishLogsServer) (err er
 			}
 		}
 
+		// if we have a close message, close out the subscription
+		if log.Close {
+			// Mark done and then set to nil so if we error after this point,
+			// we don't try to close again in the defer
+			currentSubscription.Done(remote.NodeID, err)
+			currentSubscription = nil
+			return stream.SendAndClose(&api.PublishLogsResponse{})
```

Comment:
Is this erroring out? Not sure how's the workflow flowing.

Basically, agent sends a `Close = true` then a `CloseSend`. Here we are receiving this, then sending a `PublishLogsResponse`. Is this going to be received by the agent? Errored out?

BTW this comment has nothing to do with this PR, more about my earlier code. I don't know what the flow is.